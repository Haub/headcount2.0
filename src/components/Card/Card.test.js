import React from 'react';
import { shallow, mount } from 'enzyme';
import { Card } from './index';

describe ('CARD', () => {

  let wrapper;
  beforeEach=(() => {
    wrapper=shallow(<Card 
      location = {'Academy 20'}
      stats = {{2004: 0.5}}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have class of below if stat value is below 0.5', () => {
    wrapper=shallow(<Card location={'Academy 20'} mockStats= {{ 2004: 0.3, 2005: 0.6 }} key={1} />)
    expect(wrapper.find('li').is('.below')).toEqual(true);
  })

  it('should have class of above if stat value is above 0.5', () => {
    wrapper=shallow(<Card location={'Academy 20'} mockStats= {{ 2004: 0.3, 2005: 0.6 }} key={1} />)
    expect(wrapper.find('li').is('.above')).toHaveLength(1);
  })

});