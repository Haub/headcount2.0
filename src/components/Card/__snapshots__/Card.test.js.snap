// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    location="Academy 20"
    stats={
        Object {
            "2004": 0.5,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="location"
>
          Academy 20
</h2>,
        <ul
          className="stats"
>
          <li
                    className="above"
          >
                    2004
                    : 
                    0.5
          </li>
</ul>,
      ],
      "className": "district-cards",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Academy 20",
          "className": "location",
        },
        "ref": null,
        "rendered": "Academy 20",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="above"
>
              2004
              : 
              0.5
</li>,
          ],
          "className": "stats",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2004",
                ": ",
                0.5,
              ],
              "className": "above",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              ": ",
              0.5,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="location"
>
            Academy 20
</h2>,
          <ul
            className="stats"
>
            <li
                        className="above"
            >
                        2004
                        : 
                        0.5
            </li>
</ul>,
        ],
        "className": "district-cards",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Academy 20",
            "className": "location",
          },
          "ref": null,
          "rendered": "Academy 20",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="above"
>
                2004
                : 
                0.5
</li>,
            ],
            "className": "stats",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2004",
                  ": ",
                  0.5,
                ],
                "className": "above",
              },
              "ref": null,
              "rendered": Array [
                "2004",
                ": ",
                0.5,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
